scripts/index.js

```diff
@@ -166,4 +166,51 @@ window.onload = function () {
       e.preventDefault();
     }
   });
+
+  function storageAvailable(type) {
+    let storage;
+    try {
+      storage = window[type];
+      const x = '__storage_test__';
+      storage.setItem(x, x);
+      storage.removeItem(x);
+      return true;
+    } catch (e) {
+      return e instanceof DOMException && (
+        // everything except Firefox
+        e.code === 22
+        // Firefox
+        || e.code === 1014
+        // test name field too, because code might not be present
+        // everything except Firefox
+        || e.name === 'QuotaExceededError'
+        // Firefox
+        || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+        // acknowledge QuotaExceededError only if there's something already stored
+        && (storage && storage.length !== 0);
+    }
+  }
+
+  const formFields = Array.of(...document.getElementsByClassName('form-field'));
+  let fields = { name: String, email: String, message: String };
+
+  if (storageAvailable('localStorage')) {
+    if (localStorage.getItem('fields')) {
+      fields = JSON.parse(localStorage.getItem('fields'));
+      formFields.forEach((field) => {
+        if (fields[field.id]) {
+          field.value = fields[field.id];
+        }
+      });
+    }
+  } else {
+    console.log('Local Storage not available');
+  }
+
+  formFields.forEach((field) => {
+    field.onkeydown = ({ target }) => {
+      fields[target.id] = target.value;
+      localStorage.setItem('fields', JSON.stringify(fields));
+    };
+  });
```

Comment:
- [ ] Good job team on preserving the data in the local storage. However, I noticed that after typing my message in the `textarea` input and entering my email in the `email` input, the last character of my message and email disappears after reloading the page. Please try to fix it as it will improve user experience okay 😄. Maybe try using onchange instead of `onkeydown` and see if it resolves the issue.